data-avro/src/main/java/com/linkedin/data/avro/DataTranslator.java

```diff
@@ -666,6 +645,19 @@ else if (fieldValue == null)
                 continue;
               }
             }
+
+            if ((fieldDataSchema.getDereferencedType() != DataSchema.Type.UNION) && fieldAvroSchema.getType() == Schema.Type.UNION) {
+              // Since optional field will be represented as union in Avro
```

Comment:
This comment is misleading. This case is no longer is unique to optional fields (after all, this revelation is the reason for this PR). I'd suggest just removing this line comment and keeping the comment in the line below.